src/main/java/org/prebid/server/bidder/dmx/DmxBidder.java

```diff
@@ -121,6 +126,38 @@ public DmxBidder(String endpointUrl, JacksonMapper mapper) {
                 errors);
     }
 
+    private static void validateRequest(BidRequest bidRequest) {
+        final User user = bidRequest.getUser();
+        final App app = bidRequest.getApp();
+        final Device device = bidRequest.getDevice();
+
+        if (user == null && app == null) {
+            throw new PreBidException("No user id or app id found. Could not send request to DMX.");
+        }
+
+        if (app != null) {
+            if (StringUtils.isNotBlank(app.getId())) {
+                return;
+            } else if (device != null && StringUtils.isNotBlank(device.getIfa())) {
+                return;
+            }
+        }
+
+        if (user != null) {
+            if (StringUtils.isNotBlank(user.getId())) {
+                return;
+            }
+
+            final ExtUser userExt = user.getExt();
+            // Notice that digitrust is absent to keep prebid convention
+            if (userExt != null && CollectionUtils.isNotEmpty(userExt.getEids())) {
+                return;
```

Comment:
Lets add test to check if user has id or user has ext.eids, no exception will be thrown

Reply:
https://github.com/prebid/prebid-server-java/pull/1245/files#diff-4cd61d96413fcf177895333d7291870176e282b7a6ac3031c586f1cc2a8795b3R108